File: tests/basic/statements/ConditionTest.java
package tests.basic.statements;

public class ConditionTest {

  public static void main(String[] args) {
    int a = 4;
    int b = 19;
    if (a > b) {
      System.out.println("A");
    }
    if (a < b) {
      System.out.println("B");
    }
    if (a <= b) {
      System.out.println("C");
    }
    if (a == b) {
      System.out.println("D");
    } else {
      System.out.println("E");
    }
  }
}

File: tests/basic/statements/condition.test.ts
import ClassManager from "@base/ClassLoader"
import { createNewTestClasses } from "@base/Stubs"
import TestSystem from "@stub/tests/TestSystem"
import { expect, test } from "bun:test"

test("statement: condition (if, else if)", () => {
	const className = "tests/basic/statements/ConditionTest"
	const stubClasses = createNewTestClasses()

	const classManager = new ClassManager(stubClasses)
	const system = stubClasses.getStubClass("java.lang.System") as TestSystem
	const consoleOutput = system.out

	const main = classManager.get(className)
	main.executeMethod("main", classManager)

	expect(consoleOutput.printlnLines[0]).toBe("B")
	expect(consoleOutput.printlnLines[1]).toBe("C")
	expect(consoleOutput.printlnLines[2]).toBe("E")
})
